langchain4j-vertex-ai/src/main/java/dev/langchain4j/model/vertexai/VertexAiImagenImageModel.java

```diff
@@ -0,0 +1,403 @@
+package dev.langchain4j.model.vertexai;
+
+import com.google.cloud.aiplatform.v1.EndpointName;
+import com.google.cloud.aiplatform.v1.PredictResponse;
+import com.google.cloud.aiplatform.v1.PredictionServiceClient;
+import com.google.cloud.aiplatform.v1.PredictionServiceSettings;
+import com.google.protobuf.Value;
+import com.google.protobuf.util.JsonFormat;
+import dev.langchain4j.data.image.Image;
+import dev.langchain4j.model.image.ImageModel;
+import dev.langchain4j.model.output.Response;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.google.protobuf.Value.newBuilder;
+import static dev.langchain4j.internal.Json.toJson;
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static java.util.Collections.singletonList;
+
+/**
+ * Image model for the Google Cloud Vertex AI Imagen image generation models.
+ * Supports both versions of Imagen v1 and v2, respectively identified by the model names
+ * <code>imagegeneration@002</code> and <code>imagegeneration@005</code>.
+ */
+public class VertexAiImagenImageModel implements ImageModel {
```

Comment:
It feels that "Imagen" should be removed from the class name, to be able to use it with other image models which might come later